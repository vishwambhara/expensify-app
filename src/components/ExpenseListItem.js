import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ id, createdAt, description, amount }) => (
    <div>
        <Link to = {`/edit/${id}`} >{description}</Link>
        <p>{amount} - {createdAt}</p>
    </div>
);

export default ExpenseListItem;