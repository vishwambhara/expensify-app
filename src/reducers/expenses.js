import fetch from 'isomorphic-fetch';

// Expense Reducer
const expensesReducerDefaultState = [];

const expensesReducer = (state = expensesReducerDefaultState, action) => {
    switch(action.type) {
        case 'ADD_EXPENSE': 
            // fetch('http://192.168.1.14:8080/api/v1/expenses', {
            //     method: 'POST',
            //     headers: {
            //         Accept: 'application/json',
            //         'Content-Type': 'application/json',
            //         'Access-Control-Allow-Origin': '*' 
            //     },
            //     body: JSON.stringify(action.expense),
            // });
            return [
                ...state,
                action.expense    
            ]
        case 'ADD_FROM_DATABASE':
            return [
                ...state,
                action.expense
            ]
        case 'REMOVE_EXPENSE':
            // fetch(`http://192.168.1.14:8080/api/v1/expenses/${action.id}`, {
            //     method: 'DELETE',
            //     headers: {
            //         Accept: 'application/json',
            //         'Content-Type': 'application/json',
            //         'Access-Control-Allow-Origin': '*'
            //     },
            // });
            return state.filter(({ id }) => action.id !== id);
        case 'EDIT_EXPENSE':
            return state.map((expense) => {
                if(expense.id === action.id) {
                //     fetch(`http://192.168.1.14:8080/api/v1/expenses/${action.id}`, {
                //         method: 'PUT',
                //         headers: {
                //             Accept: 'application/json',
                //             'Content-Type': 'application/json',
                //             'Access-Control-Allow-Origin': '*'
                //         },
                //         body: JSON.stringify({id: action.id, ...action.updates})
                //    });
                    return {
                        ...expense,
                        ...action.updates
                    }
                }
                else {
                    return expense;
                }
            });
        default: 
            return state;
    }
};

export default expensesReducer;