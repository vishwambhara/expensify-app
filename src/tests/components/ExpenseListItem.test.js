import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

test('should render an Expense list item', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]} />)
    expect(wrapper).toMatchSnapshot();
});