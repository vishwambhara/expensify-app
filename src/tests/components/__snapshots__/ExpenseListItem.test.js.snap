// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an Expense list item 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    Gun
  </Link>
  <p>
    195
     - 
    0
  </p>
</div>
`;
